### PR TITLE
Add ability to delete Account

### PR DESCRIPTION
**Pull request by mattia-malnis**
Issue [#571](https://github.com/maybe-finance/maybe/issues/571)

https://github.com/maybe-finance/maybe/assets/15941778/46ae4228-e795-4fb4-b727-4d38f47b38b5

### CODE REVIEW

**Comment by mattia-malnis** [comment]
I needed to change `AccountSyncJob`. When an account is deleted, we get the `ActiveJob::DeserializationError` because the job was enqueued using an account instance that could be deleted. 
By enqueuing the job with only the Account ID, the job can retrieve the Account record from the database when it's executed, avoiding deserialization errors caused by deleted records.

**Comment by mattia-malnis** [comment]
@zachgoll I'm trying to figure out how to fix the error "No live threads left. Deadlock?" and I noticed [your PR](https://github.com/maybe-finance/maybe/pull/583). 
When you'll close the PR 583 I'll remove my commit `c0e4ae8` as I didn't know how to stop this loop.

**Comment by zachgoll** [comment]
@mattia-malnis yep, I saw your PR was getting stuck on the CI tests, so opened up #583 to unblock this one. 

I _think_ you should be all set on this PR now.  Let me know if you run into issues.

**Comment by mattia-malnis** [comment]
@zachgoll thanks a million!

I should've finished with this, the only missing part is to add the confirm modal of the [PR 580](https://github.com/maybe-finance/maybe/pull/580).

Furthermore, I was thinking about on how many balances/valuations/transactions will be destroyed whan an account is deleted. Do you think it would be better to handle the destruction of these dependencies in the background?

**Comment by mattia-malnis** [comment]
@zachgoll now you can find the custom confirm modal!


https://github.com/maybe-finance/maybe/assets/15941778/83bac6df-209b-449c-b9f7-02ebd06c99ba

File: app/models/account/syncable.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ module Account::Syncable
     extend ActiveSupport::Concern
 
     def sync_later
-        AccountSyncJob.perform_later self
+        AccountSyncJob.perform_later(id)
```
Comment (zachgoll) [inline app/models/account/syncable.rb:5]: Was this a leftover commit?  Or is this still required?

Comment (mattia-malnis) [reply app/models/account/syncable.rb]: Now we have removed `after_commit :sync_account` from Transaction and Valuation so it's not necessary, I reverted the commit!

Comment (zachgoll) [reply app/models/account/syncable.rb]: Awesome, thanks!

**Pull request merged**